tools/idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/webimport/lucide/util/LruCache.kt

```diff
@@ -0,0 +1,41 @@
+package io.github.composegears.valkyrie.ui.screen.webimport.lucide.util
+
+/**
+ * Simple LRU (Least Recently Used) cache implementation with automatic eviction.
+ * When capacity is exceeded, the least recently accessed entry is automatically removed.
+ *
+ * Thread-safe operations are available via [ThreadSafeLruCache].
+ */
+class LruCache<K, V>(private val maxSize: Int) {
```

Comment:
Why not `LruCache` from androidx.collection?

<img width="785" height="896" alt="Image" src="https://github.com/user-attachments/assets/30483e13-f669-4ee3-a419-f402f44848f7" />

Reply:
Just felt it was simpler to maintain a custom one since all we need it for is entry count, rather than add the androidx dependency and not use much of its features.
I'm happy to switch to the androidx LruCache though if you think it's better here.

Reply:
Let's switch to AndroidX version